components/places/src/bookmark_sync/store.rs

```diff
@@ -85,16 +85,15 @@ impl<'a> BookmarksStore<'a> {
         for incoming in inbound.changes {
             applicator.apply_payload(incoming.0, incoming.1)?;
             incoming_telemetry.applied(1);
+            if tx.should_commit() {
+                // Trigger frecency updates for all new origins.
+                log::debug!("Updating origins for new synced URLs since last commit");
+                delete_pending_temp_tables(&self.db)?;
+            }
             tx.maybe_commit()?;
             self.interruptee.err_if_interrupted()?;
         }
 
-        // Trigger frecency updates for all new origins.
-        log::debug!("Updating origins for new incoming URLs");
-        self.interruptee.err_if_interrupted()?;
-        self.db
-            .execute_batch("DELETE FROM moz_updateoriginsinsert_temp")?;
-
         tx.commit()?;
```

Comment:
Don't we need a `delete_pending_temp_tables()` here?

Reply:
Yes! 👀